communication/slack-guidelines.md

```diff
@@ -1,90 +1,232 @@
-If it’s a code of conduct issue, please send an email to conduct@kubernetes.io and describe the situation
 
-## BOTS, TOKENS, WEBHOOKS, OH MY
+### DM (Direct Message) Conversations
 
-Bots, tokens, and webhooks are reviewed on a case-by-case basis with most requests being rejected due to security, privacy, and usability concerns. Bots and the like tend to make a lot of noise in channels. Our Slack instance has over 50,000 people and we want everyone to have a great experience. Please join #slack-admins and have a discussion about your request before requesting the access.  
+Please do not engage in proprietary company specific conversations in the
+Kubernetes Slack instance. This workspace is meant for conversations related to
+Kubernetes open source topics and community. Proprietary conversations should
+occur in your company Slack and/or communication platforms.  As with all
+communication, please be mindful of appropriateness, professionalism, and
+applicability to the Kubernetes community.
 
-Typically OK: GitHub, CNCF requests, and tools/platforms that we use to contribute to Kubernetes
 
-## ADMIN MODERATION
+### Specific Channel Rules
 
-Be mindful of how you handle communication during stressful interactions.  Administrators act as direct representatives of the community, and need to maintain a very high level of professionalism at all times. If you feel too involved in the situation to maintain impartiality or professionalism, that’s a great time to enlist the help of another admin.
+Some channels have specific rules or guidelines. If they do, they will be listed
+in the purpose or pinned docs of that channel.
 
-Try to take any situations that involve upset or angry members to DM or video chat. Please document these interactions for other Slack admins to review.
+- `#kubernetes-dev` - Questions and discourse around upstream contributions and
+development to kubernetes,
+- `#kubernetes-careers` - Job openings for positions working with/on/around
+  Kubernetes. Post the job once and pin it. Pins expire after 30 days. Postings
+  must include:
+  - A link to the posting or job description.
+  - The business name that will employ the Kubernetes hire.
+  - The location of the role or if remote is OK.
 
-Content will be automatically removed if it violates code of conduct or is a sales pitch. Admins will take a screenshot of such behavior in order to document the situation.  The community takes such violations extremely seriously, and they will be handled swiftly.
 
-## INACTIVATING ACCOUNTS
+### Escalating and/or Reporting a Problem
 
-For reasons listed below, admins may inactivate individual Slack accounts. Due to Slack’s framework, it does not allow for an account to be banned or suspended in the traditional sense. [Visit Slack’s policy on this.](https://get.Slack.help/hc/en-us/articles/204475027-Deactivate-a-member-s-account)
+Join the `#slack-admins` channel or contact one of the admins in the
```

Comment:
@mrbobbytables please, mention that DM'ing admins is not a preferred way of communication and it should be used as a last resort (e.g. - for the sensitive cases). Let's use #slack-admins as a primary communication channel.

Reply:
Updated to reflect that along with adding the new "Report message" functionalogy.